src/mono/wasm/Wasm.Build.Tests/WasmTemplateTests.cs

```diff
@@ -87,6 +87,7 @@ public void BrowserBuildThenPublish(string config)
             string projectName = Path.GetFileNameWithoutExtension(projectFile);
 
             UpdateBrowserMainJs(DefaultTargetFramework);
+            AddItemsPropertiesToProject(projectFile, "<WasmIncludeFullIcuData>true</WasmIncludeFullIcuData>");
```

Comment:
With this PR, IIUC, sharding will be the new default for !invariant case. So, that's what we should be testing here. Don't set this property.

Instead, add new tests for console/browser/blazor for `WasmIncludeFullIcuData=true`.

Reply:
The default behavior is not deterministic because it depends on the environment locale. I don't think we can let the tests to run in the default mode and hope that every time the locale will be "en" so EFIGS would be selected. Or can we?

BuildTestBase when doing `AssertBasicAppBundle` assumes that all the tests run in `FullIcu` mode by default, so requires the test-writer to add <WasmIncludeFullIcuData>true</WasmIncludeFullIcuData> property if they are overwriting the default set of `BuildArgs` record.

Reply:
`AssertBasicAppBundle` seems to be using whatever `GlobalizationMode` was passed in, based on the changes in this PR. Did I misunderstand that?

>  I don't think we can let the tests to run in the default mode and hope that every time the locale will be "en" so EFIGS would be selected.

correct. so:
- For all the non-globalization related tests set the locale explicitly, when running the apps.
- For the globalization ones, decide on a test by test basis.